### PR TITLE
Move to node 16

### LINKED ISSUE #5577: Upgrade to Node 16

### PR DESCRIPTION
**Pull request by georgeblahblah**
## What does this change?

- Moves to `node v16.19.0`, because Node 14 End of Life is in April 2023
- Updates the `ensure` script to work with `npm@8`:
  - pass `--legacy-peer-deps` so that `install` does not fail the peer dependencies check
  - pass `--package-lock=false` so that the install does not write into `yarn.lock`. This was causing a failure when later running `yarn install`
- adds error logging to the `ensure` script
- updates names of AMIgo images for both DCR and AR. The new images:
  - are based on Ubuntu 20 (Focal), as Ubuntu 18 is End of Life soon
  - no longer install `aws-security-agent`
  - install `node 16.19.0`
- uses `current` for AR's node babel-env version, which matches DCR's value
- Fixes the `build-check` script which was [failing silently](https://github.com/guardian/dotcom-rendering/actions/runs/4135033898/jobs/7146945837#step:6:12) following the webpack single entry changes

Closes #5577 

## How to test

- [x] This has been tested on CODE

### CODE REVIEW

File: dotcom-rendering/scripts/test/build-check.js [hunk 1]
```diff
@@ -41,17 +41,11 @@ const fileExists = async (glob) => {
 	}
 
 	[
-		'sentryLoader.js',
```
Comment (AshCorr) [inline dotcom-rendering/scripts/test/build-check.js:44]: Is just a driveby fix for the single bundle stuff?

**Review by AshCorr — APPROVED** [review]
👍 

Theres some features of Node 16 I'd really like to use for DCR!

File: dotcom-rendering/scripts/test/build-check.js (hunk 1 above)
Comment (georgeblahblah) [reply dotcom-rendering/scripts/test/build-check.js:44]: Oop! I should have added this to the PR description, sorry! Yes, this reflects the [single entry point](https://github.com/guardian/dotcom-rendering/pull/7086) change. This [test was failing silently](https://github.com/guardian/dotcom-rendering/actions/runs/4135033898/jobs/7146945837#step:6:12) for some time!

**Comment by arelra** [comment]
This is great! I wonder if this means we can jump to Node 18 relatively painlessly?

**Comment by bryophyta** [comment]
nice!!

**Comment by georgeblahblah** [comment]
> This is great! I wonder if this means we can jump to Node 18 relatively painlessly?

@arelra I think so! I also saw your [PR in Frontend](https://github.com/guardian/frontend/pull/25892) too which is very helpful to see what changes could be necessary here.
